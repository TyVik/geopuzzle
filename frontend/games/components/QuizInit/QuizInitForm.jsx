import {Form, FormGroup} from "react-bootstrap";
import {FormattedMessage as Msg} from "react-intl";
import React from "react";


class QuizInitForm extends React.Component {
  renderItem = (item) => {
    return <Form.Check inline
                       label={<Msg id={item}/>}
                       key={item}
                       onClick={() => this.props.toggle(item)}
                       type="checkbox"
                       defaultChecked={this.props.data[item]}/>;
  };

  render() {
    return <FormGroup className="checkbox-group" controlId="quiz">
      {this.props.available.map(this.renderItem)}
    </FormGroup>;
  }
}

export default QuizInitForm;
