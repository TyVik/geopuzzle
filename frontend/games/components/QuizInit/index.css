.checkbox-group {
    display: flex;
    justify-content: space-around;
}

.modal-backdrop {
    bottom: 0;
    z-index: 1040;
    position: fixed;
}