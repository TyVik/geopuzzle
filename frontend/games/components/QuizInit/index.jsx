'use strict';
import React from "react";
import localization from '../../../localization';
import {Button, Modal, FormGroup, Checkbox} from "react-bootstrap";
import "./index.css";


class QuizInit extends React.Component {
  constructor(props) {
    super(props);
    this.state = {title: true, flag: true, coat_of_arms: true, capital: true};
  }

  toggle(param) {
    this.setState({...this.state, [param]: !this.state[param]});
  }

  static show_checkbox(param) {
    return window.__OPTIONS__.indexOf(param) >= 0;
  }

  render_footer() {
    if (this.state.title || this.state.flag || this.state.coat_of_arms || this.state.capital) {
      return <Button onClick={() => this.props.load(this.state)}>{localization.start}</Button>;
    } else {
      return <React.Fragment>
        <i>{localization.quizInitCheck}</i>
        &nbsp;
        <Button disabled>{localization.start}</Button>
      </React.Fragment>;
    }
  }

  render() {
    return <Modal show={this.props.show} dialogClassName="custom-modal">
      <Modal.Header>
        <Modal.Title id="contained-modal-title-lg">{localization.quizInitCaption}</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <FormGroup className="checkbox-group">
          <Checkbox inline onClick={() => this.toggle('title')} defaultChecked={this.state['title']}>{localization.title}</Checkbox>
          {QuizInit.show_checkbox('flag') &&
            <Checkbox inline onClick={() => this.toggle('flag')} defaultChecked={this.state['flag']}>{localization.flag}</Checkbox>}
          {QuizInit.show_checkbox('coat_of_arms') &&
            <Checkbox inline onClick={() => this.toggle('coat_of_arms')} defaultChecked={this.state['coat_of_arms']}>{localization.coat_of_arms}</Checkbox>}
            <Checkbox inline onClick={() => this.toggle('capital')} defaultChecked={this.state['capital']}>{localization.capital}</Checkbox>
        </FormGroup>
      </Modal.Body>
      <Modal.Footer>
        {this.render_footer()}
      </Modal.Footer>
    </Modal>;
  }
}


export default QuizInit;
