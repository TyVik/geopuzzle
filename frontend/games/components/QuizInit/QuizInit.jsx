'use strict';
import React from "react";
import {Button, Modal} from "react-bootstrap";
import {FormattedMessage as Msg} from "react-intl";

import "./index.css";
import QuizInitForm from "./QuizInitForm";


class QuizInit extends React.Component {
  constructor(props) {
    super(props);
    this.state = props.options.reduce((prev, current) => {prev[current] = true; return prev;}, {});
  }

  toggle = (param) => {
    this.setState(state => ({...state, [param]: !state[param]}));
  };

  renderFooter() {
    if (this.state.name || this.state.flag || this.state.coat_of_arms || this.state.capital) {
      return <Button onClick={() => this.props.load(this.state)}><Msg id="start"/></Button>;
    } else {
      return <React.Fragment>
        <i><Msg id="quizInitCheck"/></i>
        &nbsp;
        <Button disabled={true}><Msg id="start"/></Button>
      </React.Fragment>;
    }
  }

  render() {
    return <Modal show={this.props.show} dialogClassName="custom-modal">
      <Modal.Header>
        <Modal.Title id="contained-modal-title-lg"><Msg id="quizInitCaption"/></Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <QuizInitForm available={this.props.options} data={this.state} toggle={this.toggle}/>
      </Modal.Body>
      <Modal.Footer>
        {this.renderFooter()}
      </Modal.Footer>
    </Modal>;
  }
}


export default QuizInit;
