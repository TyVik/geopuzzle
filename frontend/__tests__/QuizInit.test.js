'use strict';
import React from 'react';
import {shallow} from 'enzyme';
import QuizInit from '../games/components/QuizInit';
import {Button, Modal} from "react-bootstrap";


describe('shallow <QuizInit /> components', () => {
  let onLoad = jest.fn();
  let props = {show: true, load: onLoad};

  beforeEach(() => {
    window.__OPTIONS__ = ['name', 'flag', 'coat_of_arms', 'capital'];
  });

  it('render hidden', () => {
    expect(shallow(<QuizInit show={false}/>).html()).toBe('');
  });

  it('render', () => {
    expect(shallow(<QuizInit {...props}/>)).toMatchSnapshot('quizinit');
  });

  it('check props', () => {
    let wrapper = shallow(<QuizInit {...props}/>);
    expect(wrapper.find(Button).prop('disabled')).toBe(false);
    window.__OPTIONS__.map(key => {
      wrapper.instance().toggle(key);
    });
    expect(wrapper.find(Button).prop('disabled')).toBe(true);
    expect(wrapper.find(Modal.Footer).childAt(0).childAt(0).prop('id')).toMatch("quizInitCheck");
    wrapper.instance().toggle('title');
    wrapper.find(Button).simulate('click');
    expect(onLoad).toHaveBeenCalledWith({name: true, flag: false, coat_of_arms: false, capital: false});
  });
});
