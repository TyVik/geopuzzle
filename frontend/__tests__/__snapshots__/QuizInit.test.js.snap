// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow <QuizInit /> components check props: quizinit 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuizInit
    load={[MockFunction]}
    show={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animation": true,
      "autoFocus": true,
      "backdrop": true,
      "bsClass": "modal",
      "children": Array [
        <ModalHeader
          bsClass="modal-header"
          closeButton={false}
          closeLabel="Close"
        >
          <ModalTitle
            bsClass="modal-title"
            componentClass="h4"
            id="contained-modal-title-lg"
          >
            I'll search the regions by
          </ModalTitle>
        </ModalHeader>,
        <ModalBody
          bsClass="modal-body"
          componentClass="div"
        >
          <FormGroup
            bsClass="form-group"
            className="checkbox-group"
          >
            <Checkbox
              bsClass="checkbox"
              defaultChecked={true}
              disabled={false}
              inline={true}
              onClick={[Function]}
              title=""
            >
              Title
            </Checkbox>
            <Checkbox
              bsClass="checkbox"
              defaultChecked={true}
              disabled={false}
              inline={true}
              onClick={[Function]}
              title=""
            >
              Flag
            </Checkbox>
            <Checkbox
              bsClass="checkbox"
              defaultChecked={true}
              disabled={false}
              inline={true}
              onClick={[Function]}
              title=""
            >
              Coat of arms
            </Checkbox>
            <Checkbox
              bsClass="checkbox"
              defaultChecked={true}
              disabled={false}
              inline={true}
              onClick={[Function]}
              title=""
            >
              Capital
            </Checkbox>
          </FormGroup>
        </ModalBody>,
        <ModalFooter
          bsClass="modal-footer"
          componentClass="div"
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[Function]}
          >
            Start
          </Button>
        </ModalFooter>,
      ],
      "dialogClassName": "custom-modal",
      "dialogComponentClass": [Function],
      "enforceFocus": true,
      "keyboard": true,
      "manager": ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      },
      "onHide": [Function],
      "renderBackdrop": [Function],
      "restoreFocus": true,
      "show": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "modal-header",
          "children": <ModalTitle
            bsClass="modal-title"
            componentClass="h4"
            id="contained-modal-title-lg"
          >
            I'll search the regions by
          </ModalTitle>,
          "closeButton": false,
          "closeLabel": "Close",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-title",
            "children": "I'll search the regions by",
            "componentClass": "h4",
            "id": "contained-modal-title-lg",
          },
          "ref": null,
          "rendered": "I'll search the regions by",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "modal-body",
          "children": <FormGroup
            bsClass="form-group"
            className="checkbox-group"
          >
            <Checkbox
              bsClass="checkbox"
              defaultChecked={true}
              disabled={false}
              inline={true}
              onClick={[Function]}
              title=""
            >
              Title
            </Checkbox>
            <Checkbox
              bsClass="checkbox"
              defaultChecked={true}
              disabled={false}
              inline={true}
              onClick={[Function]}
              title=""
            >
              Flag
            </Checkbox>
            <Checkbox
              bsClass="checkbox"
              defaultChecked={true}
              disabled={false}
              inline={true}
              onClick={[Function]}
              title=""
            >
              Coat of arms
            </Checkbox>
            <Checkbox
              bsClass="checkbox"
              defaultChecked={true}
              disabled={false}
              inline={true}
              onClick={[Function]}
              title=""
            >
              Capital
            </Checkbox>
          </FormGroup>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Title
              </Checkbox>,
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Flag
              </Checkbox>,
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Coat of arms
              </Checkbox>,
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Capital
              </Checkbox>,
            ],
            "className": "checkbox-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "checkbox",
                "children": "Title",
                "defaultChecked": true,
                "disabled": false,
                "inline": true,
                "onClick": [Function],
                "title": "",
              },
              "ref": null,
              "rendered": "Title",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "checkbox",
                "children": "Flag",
                "defaultChecked": true,
                "disabled": false,
                "inline": true,
                "onClick": [Function],
                "title": "",
              },
              "ref": null,
              "rendered": "Flag",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "checkbox",
                "children": "Coat of arms",
                "defaultChecked": true,
                "disabled": false,
                "inline": true,
                "onClick": [Function],
                "title": "",
              },
              "ref": null,
              "rendered": "Coat of arms",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "checkbox",
                "children": "Capital",
                "defaultChecked": true,
                "disabled": false,
                "inline": true,
                "onClick": [Function],
                "title": "",
              },
              "ref": null,
              "rendered": "Capital",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "modal-footer",
          "children": <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[Function]}
          >
            Start
          </Button>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Start",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Start",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": true,
        "autoFocus": true,
        "backdrop": true,
        "bsClass": "modal",
        "children": Array [
          <ModalHeader
            bsClass="modal-header"
            closeButton={false}
            closeLabel="Close"
          >
            <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title-lg"
            >
              I'll search the regions by
            </ModalTitle>
          </ModalHeader>,
          <ModalBody
            bsClass="modal-body"
            componentClass="div"
          >
            <FormGroup
              bsClass="form-group"
              className="checkbox-group"
            >
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Title
              </Checkbox>
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Flag
              </Checkbox>
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Coat of arms
              </Checkbox>
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Capital
              </Checkbox>
            </FormGroup>
          </ModalBody>,
          <ModalFooter
            bsClass="modal-footer"
            componentClass="div"
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
            >
              Start
            </Button>
          </ModalFooter>,
        ],
        "dialogClassName": "custom-modal",
        "dialogComponentClass": [Function],
        "enforceFocus": true,
        "keyboard": true,
        "manager": ModalManager {
          "add": [Function],
          "containers": Array [],
          "data": Array [],
          "handleContainerOverflow": true,
          "hideSiblingNodes": true,
          "isTopModal": [Function],
          "modals": Array [],
          "remove": [Function],
        },
        "onHide": [Function],
        "renderBackdrop": [Function],
        "restoreFocus": true,
        "show": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-header",
            "children": <ModalTitle
              bsClass="modal-title"
              componentClass="h4"
              id="contained-modal-title-lg"
            >
              I'll search the regions by
            </ModalTitle>,
            "closeButton": false,
            "closeLabel": "Close",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "modal-title",
              "children": "I'll search the regions by",
              "componentClass": "h4",
              "id": "contained-modal-title-lg",
            },
            "ref": null,
            "rendered": "I'll search the regions by",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-body",
            "children": <FormGroup
              bsClass="form-group"
              className="checkbox-group"
            >
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Title
              </Checkbox>
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Flag
              </Checkbox>
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Coat of arms
              </Checkbox>
              <Checkbox
                bsClass="checkbox"
                defaultChecked={true}
                disabled={false}
                inline={true}
                onClick={[Function]}
                title=""
              >
                Capital
              </Checkbox>
            </FormGroup>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <Checkbox
                  bsClass="checkbox"
                  defaultChecked={true}
                  disabled={false}
                  inline={true}
                  onClick={[Function]}
                  title=""
                >
                  Title
                </Checkbox>,
                <Checkbox
                  bsClass="checkbox"
                  defaultChecked={true}
                  disabled={false}
                  inline={true}
                  onClick={[Function]}
                  title=""
                >
                  Flag
                </Checkbox>,
                <Checkbox
                  bsClass="checkbox"
                  defaultChecked={true}
                  disabled={false}
                  inline={true}
                  onClick={[Function]}
                  title=""
                >
                  Coat of arms
                </Checkbox>,
                <Checkbox
                  bsClass="checkbox"
                  defaultChecked={true}
                  disabled={false}
                  inline={true}
                  onClick={[Function]}
                  title=""
                >
                  Capital
                </Checkbox>,
              ],
              "className": "checkbox-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "checkbox",
                  "children": "Title",
                  "defaultChecked": true,
                  "disabled": false,
                  "inline": true,
                  "onClick": [Function],
                  "title": "",
                },
                "ref": null,
                "rendered": "Title",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "checkbox",
                  "children": "Flag",
                  "defaultChecked": true,
                  "disabled": false,
                  "inline": true,
                  "onClick": [Function],
                  "title": "",
                },
                "ref": null,
                "rendered": "Flag",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "checkbox",
                  "children": "Coat of arms",
                  "defaultChecked": true,
                  "disabled": false,
                  "inline": true,
                  "onClick": [Function],
                  "title": "",
                },
                "ref": null,
                "rendered": "Coat of arms",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "checkbox",
                  "children": "Capital",
                  "defaultChecked": true,
                  "disabled": false,
                  "inline": true,
                  "onClick": [Function],
                  "title": "",
                },
                "ref": null,
                "rendered": "Capital",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "modal-footer",
            "children": <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
            >
              Start
            </Button>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "default",
              "children": "Start",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Start",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
